components/AddTestimony/AddTestimony.js

```diff
@@ -12,7 +13,16 @@ const AddTestimony = ({
 }) => {
   const [showTestimony, setShowTestimony] = useState(false)
 
-  const handleShowTestimony = () => setShowTestimony(true)
+  const router = useRouter()
+
+  const handleShowTestimony = () => {
+    if (!authenticated) {
+      router.push("/login")
```

Comment:
We can redirect the user back to the bill page after they login by setting the `r` query parameter in the login URL, as we do [here](https://github.com/codeforboston/advocacy-maps/blob/b3fa5b09d15ccd05a4ffff532bf0fdb04e7ade11/components/auth/context.tsx#L59). 

```
        router.push({ pathname: "/login", query: { r: router.asPath } })
```

Reply:
Ahh I see!  Thank you.  I made the requested changes.